### PR TITLE
Refactor middlewares

### PR DESCRIPTION
**Pull request by mnmallea**
## Summary
- Using schema validations on users POST body.

## Known Issues
None

## Trello Card
https://trello.com/c/fYYYZqY2/27-middlewares

### CODE REVIEW

File: app/middlewares/users.js [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
+    },
+    password: {
+      errorMessage: 'password must have at least 7 characters',
+      isLength: { options: { min: 7 } }
+    },
+    email: {
+      errorMessage: 'email must have email format',
+      isEmail: true
+    }
+  })
```
Comment (lnmedrano) [inline app/middlewares/users.js:25]: Move this to a `schemas` folder with an appropriate name.

File: app/middlewares/users.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-  body('first_name', 'first_name must be present')
-    .not()
-    .isEmpty(),
-  body('last_name', 'last_name must be present')
-    .not()
-    .isEmpty(),
-  body('password', 'password must have at least 7 characters').isLength({ min: 7 }),
-  body('email', 'email must have email format').isEmail()
-]);
+exports.userBodyValidations = fieldsValidation(
```
Comment (lnmedrano) [inline app/middlewares/users.js:7]: I would remove this from users validation, and call the `fields_validations` function from the routes. Something like:
`app.post(..., [fieldsValidation(schema), ...], ...)`

File: app/middlewares/fields_validation.js [hunk 3]
```diff
@@ -8,8 +8,8 @@ const buildErrorMessage = errors =>
     .map(error => error.msg)
     .join('; ');
 
-exports.fieldsValidation = validations => (req, res, next) => {
-  Promise.all(validations.map(validation => validation.run(req)))
+exports.fieldsValidation = (...validations) => (req, res, next) => {
+  Promise.all(validations.flat().map(validation => validation.run(req)))
```
Comment (lnmedrano) [inline app/middlewares/fields_validation.js:12]: There is no need to run the schemas by hand. `checkSchema` runs the validations and let you check the errors with the function `validationResult`.

Comment (mnmallea) [reply app/middlewares/fields_validation.js:12]: If I don't run validations, when checking `validationsResult`, errors is always empty

Comment (mnmallea) [reply app/middlewares/fields_validation.js:12]: The idea of this method is to short-circuit the middleware chain calling the error middleware when validation result is fail.
If I don't do this, I will always repeat `if(!errors.isEmpty()) return next(<error>)` on all routes.

**Pull request merged**